src/plots/gl3d/scene.js

```diff
@@ -132,20 +133,34 @@ proto.tryCreatePlot = function() {
         opts.canvas = STATIC_CANVAS;
     }
 
-    var failed = 0;
+    return opts;
+};
+
+proto.tryCreatePlot = function() {
+    var scene = this;
+
+    var opts = scene.prepareOptions();
+
+    var success = true;
 
     try {
         scene.glplot = createPlot(opts);
     } catch(e) {
-        failed++;
-        try { // try second time to fix issue with Chrome 77 https://github.com/plotly/plotly.js/issues/4233
-            scene.glplot = createPlot(opts);
-        } catch(e) {
-            failed++;
+        if(scene.staticMode) {
+            success = false;
+        } else { // try second time
+            try {
+                // invert preserveDrawingBuffer setup which could be resulted from is-mobile not detecting the right device
+                isMobile = opts.glOptions.preserveDrawingBuffer = !opts.glOptions.preserveDrawingBuffer;
```

Comment:
Hmm, I'm not sure I like this. 

If the "main" problem here is `is-mobile` not doing its job in some cases, then we should try to fix `is-mobile` instead of working around it behind the scenes.

At the very (very) least we should `Lib.warn` something here to let the user (and the devs) know what's going on, and gather some data on the situations where `is-mobile` is buggy.

----

To be more robust, what if instead of flipping `preserveDrawingBuffer` here, we allow users (via a new config option) to set `preserveDrawingBuffer` to `true` to essentially override `is-mobile` when they want?

Reply:
> To be more robust, what if instead of flipping `preserveDrawingBuffer` here, we allow users (via a new config option) to set `preserveDrawingBuffer` to `true` to essentially override `is-mobile` when they want?

The users may not know in advance the correct setup for the devices.

Reply:
> Hmm, I'm not sure I like this.
> 
> If the "main" problem here is `is-mobile` not doing its job in some cases, then we should try to fix `is-mobile` instead of working around it behind the scenes.
 
> At the very (very) least we should `Lib.warn` something here to let the user (and the devs) know what's going on, and gather some data on the situations where `is-mobile` is buggy.

Good call. Done in https://github.com/plotly/plotly.js/pull/4549/commits/a26e5e2262437a96cad71da9cc4b2177064b1334.

Reply:
> The users may not know in advance the correct setup for the devices.

Yes, I know that. But do you agree that the best way to fix this problem would be to _fix_ `is-mobile`?

Reply:
.... but now with the `Lib.warn` message I guess we can expect some users to tell us about it - which may lead us to fix `is-mobile` for these cases. 

I'm happy with this solution. That `Lib.warn` message is very clear.

Reply:
I totally agree we should try fixing `is-mobile` to help detect the right device.
Still this PR helps detect/by-pass such edge cases.